deltacat/storage/rivulet/reader/dataset_metastore.py

```diff
@@ -37,29 +43,67 @@ def generate_sstables(self) -> Generator[SSTable, None, None]:
 
 
 class DatasetMetastore:
-    """Metastore implementation for manifests stored on a filesystem"""
+    """
+    Metastore implementation for manifests stored on a filesystem
+
+    TODO this will be replaced with Deltacat Storage interface - https://github.com/ray-project/deltacat/issues/477
+    """
 
     def __init__(
         self,
+        # URI at which we expect to find deltas
+        delta_root_uri: str,
         file_provider: FileProvider,
+        *,
         manifest_io: ManifestIO = None,
         sst_reader: SSTReader = None,
     ):
         self._min_key = None
         self._max_key = None
+        self.delta_root_uri = delta_root_uri
         self.file_provider = file_provider
-        self.manifest_io = manifest_io or JsonManifestIO()
+        self.manifest_io = manifest_io or DeltacatManifestIO(delta_root_uri)
         self.sst_reader = sst_reader or JsonSstReader()
 
+    def _get_delta(self, delta_dir: str, filesystem: FileSystem) -> RivuletDelta:
+        """
+        Given a DeltaCat delta directory, find latest delta file
+
+        This will be replaced by deltacat storage API.
+        Current implementation does not respect open/closed transactions, it just
+            looks for the latest revision
+        """
+        rev_directory = os.path.join(delta_dir, "rev")
+        revisions = filesystem.get_file_info(fs.FileSelector(rev_directory))
+        # Take lexicographical max
+        latest_revision = None
+        for revision in revisions:
+            latest_revision = (
+                revision if not latest_revision else max(latest_revision, revision.path)
```

Comment:
Shouldn't this be

```suggestion
                revision.path if not latest_revision else max(latest_revision, revision.path)
```

Otherwise you wouldn't be comparing the paths themselves.

Reply:
revision in this case is a path since it is generated by filesystem list